langchain-core/src/callbacks/manager.ts

```diff
@@ -20,6 +20,19 @@ import { consumeCallback } from "./promises.js";
 import { Serialized } from "../load/serializable.js";
 import type { DocumentInterface } from "../documents/document.js";
 
+if (
+  getEnvironmentVariable("LANGCHAIN_TRACING_V2") === "true" &&
+  getEnvironmentVariable("LANGCHAIN_CALLBACKS_BACKGROUND") !== "true"
+) {
+  console.warn(
+    [
+      "[WARN]: You have enabled LangSmith tracing without backgrounding callbacks.",
+      "[WARN]: If you are not using a serverless environment where you must wait for tracing calls to finish,",
+      `[WARN]: we suggest setting "process.env.LANGCHAIN_CALLBACKS_BACKGROUND=true" to avoid additional latency.`,
```

Comment:
Makes sense. We may want to do this in the `traceable` HOF in the SDK as well

Reply:
Yes let's do it